### DISCUSSION TITLE
[SOLVED] What is the "Shrink database file"?

### DISCUSSION THREAD

**Discussion by sakkamade** [martinrotter/rssguard#423] [Q&A] [answered]
I could not come across the precise definition in [docs](https://github.com/martinrotter/rssguard/blob/master/resources/docs/Documentation.md), but I fancy to understand what  exactly is
going on when user availing themself of this option. 

In particular, what are chances of important data to be removed?

P.s. This question is not much out of my curiosity as this option selects itself after click on "OK" in "Cleanup database" when 
any—and for example, only one—but this option had been selected. I am not sure if this is actual issue or by design. 
P.p.s Otherwise, if it is required to be enabled always, append some little description embedded in 
parentheses, or tooltip, between the lines: "Required to shrink the size of the database when removing messages."

**Comment by martinrotter** [comment, marked as answer]
RSS Guard uses SQLite as database. SQLite stores all data in a single file and, over time, the file naturally increases as more and more messages are stored in it.

However, if you purge some messages from database via "Database cleanup" dialog, SQLite will not automatically resize the file and will keep its previous (big) size, because of performance reasons.

So if your DB had 100 MB before cleanup and 99% of messages were removed, then your DB will still have 100 MB after the operation.

Option "shrink" will force SQLite to remove all "empty" parts of file physically from filesystem and make the file only large enough as really needed.

https://sqlite.org/lang_vacuum.html

**Reply by sakkamade** [reply]
The answer to this question
> In particular, what are chances of important data to be removed?

is: "None." then?

**Reply by martinrotter** [reply]
Yes, shrinking has like 0,01% chance of breaking something and even if power outage kicks in, SQLite creates file backup before shrinking anyway.

**Comment by martinrotter** [comment]
Also, I will check the dialog behavior. "shrink" should be optional of course.

**Reply by sakkamade** [reply]
I see, now. I will elaborate about it in the separate issue then.

**Reply by sakkamade** [reply]
Optional, but very much recommended, and should be marked as such.
Mostly the same as required, actually.
